gnucash/report/report-gnome/dialog-report-column-view.c

```diff
@@ -147,26 +149,37 @@ update_display_lists(gnc_column_view_edit * view)
     {
         for (i = 0; !scm_is_null(rpt_guids); rpt_guids = SCM_CDR(rpt_guids), i++)
         {
-            if (scm_is_equal (SCM_CAR(rpt_guids), selection))
-                row = i;
+            guid_str = scm_to_utf8_string (SCM_CAR(rpt_guids));
             name = gnc_scm_to_utf8_string (scm_call_2(template_menu_name, SCM_CAR(rpt_guids),
                                              SCM_BOOL_F));
 
             gtk_list_store_append(store, &iter);
             gtk_list_store_set(store, &iter,
                                AVAILABLE_COL_NAME, _(name),
-                               AVAILABLE_COL_ROW, i,
+                               AVAILABLE_COL_GUID, guid_str,
                                -1);
+
+            if (scm_is_equal (SCM_CAR(rpt_guids), selection))
+                gtk_tree_selection_select_iter (tree_selection, &iter);
+
             g_free (name);
+            g_free (guid_str);
         }
-
     }
 
-    tree_selection = gtk_tree_view_get_selection(view->available);
-    path = gtk_tree_path_new_from_indices(row, -1);
-    gtk_tree_selection_select_path(tree_selection, path);
-    gtk_tree_path_free(path);
-
+    /* Scroll to selection or on first load select first entry */
+    if (gtk_tree_selection_get_selected(tree_selection, &model, &iter))
+    {
+        GtkTreePath *path = gtk_tree_model_get_path (model, &iter);
+        gtk_tree_view_scroll_to_cell (view->available, path, NULL, TRUE, 0.5, 0.0);
+        gtk_tree_path_free (path);
+    }
+    else
+    {
+         gboolean valid = gtk_tree_model_get_iter_first (model, &iter);
+         if (valid)
+             gtk_tree_selection_select_iter (tree_selection, &iter);
+    }
```

Comment:
Huh, this is a bit unexpected. Why do you want to set a selection on first run ? The odds the user wants to add the first report right away are small, so the user will most likely have to select a report anyway.

As for making the selection centered in the available list, I appreciate you consider this. However I believe it's a bit too intrusive and it actually creates a variant of what the bug was opened for. For example if the selection in the available list is not exactly in the center of the treeview, a click on move up or move down will also affect the appearance of the available list as the list would recenter the selection. That is at best distracting and at worst not what the user wants at all (maybe he deliberately navigated the available list to see certain reports).

The behaviour I would expect is that if I don't touch anything in the available list nothing changes visually to it either. And the same should go for the content list.

I had thought about mentioning this in my previous review, but omitted it in the end: one way to handle this is to update the available list only when really needed (which would mean splitting the update function in two separate ones and only calling either when something affects them). The tricky part is then how to detect when the user has added/deleted/renamed a saved report configuration (directly in gnucash, not in the multicolumn report options) while a multicolumn report options dialog is still open. Ideally the available list in the open options dialog would then get updated. This is currently happening because we update the list with each report action (a bit more than needed). We could also relax this a bit and expect the user to close the options dialog and reopen it to detect new reports. It would be a less common case after all.

The alternative approach could be to save the scrolled state at the beginning of the update lists function and after filling the treeview, set the scrolled state to what was saved. This approach is closer to what you have implemented here and would equally prevent unexpected visual changes in the available list.

Reply:
The original had it selected by default but am not fussed so removed it and also set the contents tree view to also be non selected on first run which seems to work quite well with the button sensitivity changes.